mitodl/mit-open
# input / search input styling updates

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5386

### Description (What does it do?)
This PR makes changes to our `Input` element to match some updates from Figma. We now have 4 sizes of `Input`:

 - small
 - medium
 - large
 - hero

The general style of the `Input` control was updated to match the changes to the search input, and `SearchInput` was moved into `ol-components`. The `SearchInput` component implements `Input` at the `hero` size on the home page and the `large` size on the search page and channel pages. 

### Screenshots (if appropriate):
![Screenshot 2024-09-11 at 4 28 24 PM](https://github.com/user-attachments/assets/b65f0d14-f107-49d7-8572-7851e964efbd)
![Screenshot 2024-09-11 at 4 29 08 PM](https://github.com/user-attachments/assets/b3e8fb74-7ff8-41a7-8215-de5974e750ba)
![Screen Shot 2024-09-11 at 16 28 44](https://github.com/user-attachments/assets/93126e95-9200-4834-8a35-9ed6346baaf6)
![Screen Shot 2024-09-11 at 16 29 24](https://github.com/user-attachments/assets/21bac98d-3590-4c53-b91a-87c7699f2aae)

### How can this be tested?
 - Spin up `mit-learn` on this branch
 - Make sure you have some data populated in your database and a search index generated
 - Visit http://localhost:8062/search/
 - Verify that the search box behaves as expected and is properly sized
 - Visit http://localhost:8062/topics/ and choose a topic by clicking on it
 - On the topic page, verify that the search box behaves as expected and is properly sized
 - Visit the storybook page for the `Input` element and verify that everything looks right and matches Figma: https://storybook.learn.c4103.com/?path=/story/smoot-design-input--adornments
